docs/usage.rst

```diff
@@ -333,25 +333,73 @@ configuration.
 Debugging Python functions
 --------------------------
 
-Unlike Node.JS and Java, Python requires you to enable remote debugging
-in your Lambda function code. If you enable debugging with
-``--debug-port`` or ``-d`` for a function that uses one of the Python
-runtimes, SAM CLI will just map through that port from your host machine
-through to the Lambda runtime container. You will need to enable remote
-debugging in your function code. To do this, use a python package such
-as `remote-pdb <https://pypi.python.org/pypi/remote-pdb>`__. When
-configuring the host the debugger listens on in your code, make sure to
-use ``0.0.0.0`` not ``127.0.0.1`` to allow Docker to map through the
-port to your host machine.
-
-   Please note, due to a `open
-   bug <https://github.com/Microsoft/vscode-python/issues/71>`__ with
-   Visual Studio Code, you may get a
-   ``Debug adapter process has terminated unexpectedly`` error when
-   attempting to debug Python applications with this IDE. Please track
-   the `GitHub
-   issue <https://github.com/Microsoft/vscode-python/issues/71>`__ for
-   updates.
+Python debugging requires you to enable remote debugging in your Lambda function code, therefore it's a 2-step process:
+
+1. Install ptvsd library and enable within your code
+2. Configure your IDE to connect to the debugger you configured for your function
+  - SAM CLI debug port option ``--debug-port`` or `-d` will map that port to the local Lambda container execution your IDE needs to connect to
+
+Ptvsd configuration
+^^^^^^^^^^^^^^^^^^^
+
+Provided that your source code has ptvsd available for your Lambda function you can do the following to enable the debugger:
+
+.. code:: python
+
+  import ptvsd
+  ...
+
+  def lambda_handler(event, context):
+    ...
+    ptvsd.enable_attach(address=('0.0.0.0', 5890), redirect_output=True)

```

Comment:
From VSCode's Remote Python debugging [docs](https://code.visualstudio.com/docs/python/debugging): 
```
Local computer: set a breakpoint in the code where you want to start debugging.

Tip: setting a single breakpoint on the statement immediately following the ptvsd.wait_for_attach() line may not work. Set at least one other breakpoint on another statement.
```

Putting this into the function code risks some customer confusion on debugging not working correctly. Can we move this up to where ptvsd is imported?

Reply:
Just tested and it also works in the global space. Fixed:

```python
import ptvsd

# Enable ptvsd on 0.0.0.0 address and on port 5890 that we'll connect later with our IDE
ptvsd.enable_attach(address=('0.0.0.0', 5890), redirect_output=True)
ptvsd.wait_for_attach()
```